src/public/js/rs.js

```
@@ -21,6 +105,94 @@ var RS = (function() {
+      return null;
+    }
+    errors = false;
+    for (i = 0; i < n-k; i++) {
+      if (remainder[i] != 0) {
+        errors = true
+      }
+    }
+    if (errors) {
+      // TODO ERROR CORRECTION
+      return null
+    } else {
+      msg = new Uint8Array(k)
+      for (i = 0; i < k; i++) {
+        msg[i] = codedMsg[i]
+      }
+      return msg
+    }
+  }
+
+  function polynomialDiv(divisor, dividend, n, k) {
+    if (divisor.length != k || dividend.length != n) {
+      return null;
+    }
+    // precompute divisors highest order inverse and distribute to all of divisor
+    invVal = multInv[divisor[0]];
+    for (i = 0; i < k; i++){
+      divisor[i] = fieldMultiply(divisor[i], invVal);
+    }
+    for (i = 0; i < n-k; i++) {
+      // find how many of highest order of scaled divisor go in to
+      // current highest order of leftover dividend
+      // Because of precompute it is simply the coefficient of the highest order of dividend
+      quotient = dividend[i];
+
+      // find scaled divisor to subtract from current remainder of dividend
+      // by multiplying each coefficient of precomputed divisor by current
+      // highest order quotient
+      scaledDivisor = new Uint8Array(k);
+      for (j = 0; j < k; j++) {
+        scaledDivisor[j] = fieldMultiply(divisor[j], quotient);
+      }
+
+      // Subtract scaled divisor from current remainder of dividend
+      // to get new remainder of dividend which will be 1 order less
+      for (j = i; j < i+k; j++) {
+        // Galois field subtraction is xor with base field 2
+        dividend[j] = dividend[j] ^ scaledDivisor[j-i];
+      }
+    }
+
+    // I am sure there is a better way to do this but I am bad.
+    remainder = new Uint8Array(n-k-1);
+    for (i = 0; i < n - k - 1; i++) {
+      remainder[i] = dividend[i + k + 1];
+    }
+    // Adding this as kinda failsafe test case
+    for (i = 0; i < k + 1; i++) {
+      if (dividend[i] != 0) {
+        return null;
```

throw error instead of returning null?